File: 01-Languages/Go/15-Modules-and-Packages/main.go
// main.go - imports a package from a different directory within the same module.
package main

import (
	"fmt"

	"example.com/modulesdemo/mathutils"
)

func main() {
	fmt.Println("mathutils.Add(2, 3):", mathutils.Add(2, 3))
	fmt.Println("mathutils.Multiply(4, 5):", mathutils.Multiply(4, 5))
}

File: 01-Languages/Java/15-Modules-and-Packages/Main.java
// Main.java - the entry point, importing a class from a DIFFERENT file, in a DIFFERENT package,
// which must live in a matching com/example/utils/ subdirectory or this fails to compile.

import com.example.utils.MathHelpers;

public class Main {
    public static void main(String[] args) {
        System.out.println("MathHelpers.add(2, 3): " + MathHelpers.add(2, 3));
        System.out.println("MathHelpers.multiply(4, 5): " + MathHelpers.multiply(4, 5));
    }
}
